rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml

```diff
@@ -0,0 +1,147 @@
+  - **Look for Anomalous Patterns**: Determine if this actor or resource has performed similar discovery actions previously, or if these actions coincide with other alerts related to credential use or privilege escalation.
+
+### False Positive Analysis
+
+- **Expected Discovery Activity**: Regular discovery or enumeration API calls may be conducted by security, automation, or monitoring scripts to maintain an inventory of resources. Validate if this activity aligns with known automation or inventory tasks.
+- **Routine Admin or Automated Access**: If specific roles or resources, such as automation tools or monitoring services, regularly trigger this rule, consider adding exceptions for these known, benign users to reduce false positives.
+
+### Response and Remediation
+
+- **Confirm Authorized Access**: If the discovery activity appears unauthorized, consider immediate steps to restrict the user or resource’s permissions.
+- **Review and Remove Unauthorized API Calls**: If the actor is not authorized to perform discovery actions, investigate and potentially disable their permissions or access keys to prevent further misuse.
+- **Enhance Monitoring for Discovery Patterns**: Consider additional logging or alerting for high-frequency discovery API calls, especially if triggered from new or unrecognized resources.
+- **Policy Review and Updates**: Review IAM policies associated with the actor, ensuring restrictive permissions and MFA enforcement where possible to prevent unauthorized discovery.
+
+### Additional Information
+
+For further guidance on AWS infrastructure discovery and best practices, refer to [AWS CloudTrail documentation](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-event-reference.html) and MITRE ATT&CK’s [Cloud Infrastructure Discovery](https://attack.mitre.org/techniques/T1580/).
+"""
+references = ["https://stratus-red-team.cloud/attack-techniques/AWS/aws.discovery.ec2-enumerate-from-instance/"]
+risk_score = 21
+rule_id = "74f45152-9aee-11ef-b0a5-f661ea17fbcd"
+severity = "low"
+tags = [
+    "Domain: Cloud",
+    "Data Source: AWS",
+    "Data Source: AWS EC2",
+    "Data Source: AWS IAM",
+    "Data Source: AWS S3",
+    "Use Case: Threat Detection",
+    "Tactic: Discovery",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-aws.cloudtrail*
+
+// create time window buckets of 10 seconds
+| eval time_window = date_trunc(10 seconds, @timestamp)
+| where
+    event.dataset == "aws.cloudtrail"
+
+    // filter on CloudTrail audit logs for IAM, EC2, and S3 events only
+    and event.provider in ("iam.amazonaws.com", "ec2.amazonaws.com", "s3.amazonaws.com")
+
+    // ignore AWS service actions
+    and aws.cloudtrail.user_identity.type != "AWSService"
+
+    // filter for aws-cli specifically
+    and user_agent.name == "aws-cli"
```

Comment:
Why just the cli? what about `Boto3` user-agent for python scripts?

Reply:
SDK's require fetching credentials and adding them into a script or tool of sorts.
AWS CLI can be OOTB and prime for opportunistic TAs that just landed on an EC2 instance box.

Both are plausible, but the SDK route requires additional steps and alerts would likely be FP from admins or expected automation or scheduled jobs.

SCARLETEEL, Midnight Blizzard and Scattered Spider were known to use the AWS CLI installed on the compromised instance for discovery and enumeration.